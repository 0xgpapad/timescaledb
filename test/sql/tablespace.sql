
\set ON_ERROR_STOP 1
\set SHOW_CONTEXT never

\ir include/create_single_db.sql

\set ECHO ALL
\c single

\set ON_ERROR_STOP 0

SET client_min_messages = ERROR;
drop tablespace if exists tspace1;
SET client_min_messages = NOTICE;

\set VERBOSITY verbose

--test hypertable with tables space
create tablespace tspace1 location :TEST_TABLESPACE_PATH;
create table test_tspace(time timestamp, temp float, device_id text) tablespace tspace1;
select create_hypertable('test_tspace', 'time', 'device_id');
select * from _timescaledb_catalog.partition p INNER JOIN _timescaledb_catalog.partition_replica pr ON (pr.partition_id = p.id);
insert into test_tspace values ('2017-01-20T09:00:01', 24.3, 'dev1');
insert into test_tspace values ('2017-01-20T09:00:02', 22.3, 'dev7');
\dt test_tspace

--verify that the table chunk has the correct tablespace
\d+ _timescaledb_internal.*

--cleanup
\set VERBOSITY default
drop table test_tspace;
drop tablespace tspace1;
