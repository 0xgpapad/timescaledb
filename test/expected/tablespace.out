\set ON_ERROR_STOP 1
\set SHOW_CONTEXT never
\ir include/create_single_db.sql
SET client_min_messages = WARNING;
DROP DATABASE IF EXISTS single;
SET client_min_messages = NOTICE;
CREATE DATABASE single;
\c single
CREATE EXTENSION IF NOT EXISTS iobeamdb CASCADE;
psql:include/create_single_db.sql:7: NOTICE:  installing required extension "dblink"
psql:include/create_single_db.sql:7: NOTICE:  installing required extension "postgres_fdw"
psql:include/create_single_db.sql:7: NOTICE:  installing required extension "hstore"
SELECT setup_single_node(hostname => 'fakehost'); -- fakehost makes sure there is no network connection
 setup_single_node 
-------------------
 
(1 row)

\set ECHO ALL
\c single
\set ON_ERROR_STOP 0
SET client_min_messages = WARNING;
drop tablespace if exists tspace1;
SET client_min_messages = NOTICE;
\set VERBOSITY verbose
--test hypertable with tables space
create tablespace tspace1 location :TEST_TABLESPACE_PATH;
create table test_tspace(time timestamp, temp float, device_id text) tablespace tspace1;
select create_hypertable('test_tspace', 'time', 'device_id');
 create_hypertable 
-------------------
 
(1 row)

select * from _iobeamdb_catalog.partition p INNER JOIN _iobeamdb_catalog.partition_replica pr ON (pr.partition_id = p.id);
 id | epoch_id | keyspace_start | keyspace_end | tablespace | id | partition_id | hypertable_id | replica_id |    schema_name     |       table_name       
----+----------+----------------+--------------+------------+----+--------------+---------------+------------+--------------------+------------------------
  1 |        1 |              0 |        32767 | tspace1    |  1 |            1 |             1 |          0 | _iobeamdb_internal | _hyper_1_1_0_partition
(1 row)

insert into test_tspace values ('2017-01-20T09:00:01', 24.3, 'dev1');
insert into test_tspace values ('2017-01-20T09:00:02', 22.3, 'dev7');
\dt test_tspace
            List of relations
 Schema |    Name     | Type  |  Owner   
--------+-------------+-------+----------
 public | test_tspace | table | postgres
(1 row)

--verify that the table chunk has the correct tablespace
\d+ _iobeamdb_internal.*
                        Table "_iobeamdb_internal._hyper_1_0_replica"
  Column   |            Type             | Modifiers | Storage  | Stats target | Description 
-----------+-----------------------------+-----------+----------+--------------+-------------
 time      | timestamp without time zone |           | plain    |              | 
 temp      | double precision            |           | plain    |              | 
 device_id | text                        |           | extended |              | 
Inherits: _iobeamdb_internal._hyper_1_root
Child tables: _iobeamdb_internal._hyper_1_1_0_partition

                       Table "_iobeamdb_internal._hyper_1_1_0_1_data"
  Column   |            Type             | Modifiers | Storage  | Stats target | Description 
-----------+-----------------------------+-----------+----------+--------------+-------------
 time      | timestamp without time zone |           | plain    |              | 
 temp      | double precision            |           | plain    |              | 
 device_id | text                        |           | extended |              | 
Check constraints:
    "partition" CHECK (_iobeamdb_catalog.get_partition_for_key(device_id, 32768) >= '0'::smallint AND _iobeamdb_catalog.get_partition_for_key(device_id, 32768) <= '32767'::smallint)
Inherits: _iobeamdb_internal._hyper_1_1_0_partition
Tablespace: "tspace1"

                      Table "_iobeamdb_internal._hyper_1_1_0_partition"
  Column   |            Type             | Modifiers | Storage  | Stats target | Description 
-----------+-----------------------------+-----------+----------+--------------+-------------
 time      | timestamp without time zone |           | plain    |              | 
 temp      | double precision            |           | plain    |              | 
 device_id | text                        |           | extended |              | 
Check constraints:
    "partition" CHECK (_iobeamdb_catalog.get_partition_for_key(device_id, 32768) >= '0'::smallint AND _iobeamdb_catalog.get_partition_for_key(device_id, 32768) <= '32767'::smallint)
Inherits: _iobeamdb_internal._hyper_1_0_replica
Child tables: _iobeamdb_internal._hyper_1_1_0_1_data

                          Table "_iobeamdb_internal._hyper_1_root"
  Column   |            Type             | Modifiers | Storage  | Stats target | Description 
-----------+-----------------------------+-----------+----------+--------------+-------------
 time      | timestamp without time zone |           | plain    |              | 
 temp      | double precision            |           | plain    |              | 
 device_id | text                        |           | extended |              | 
Child tables: _copy_temp_1,
              _iobeamdb_internal._hyper_1_0_replica

--cleanup
drop table test_tspace;
NOTICE:  00000: drop cascades to 4 other objects
DETAIL:  drop cascades to table _iobeamdb_internal._hyper_1_0_replica
drop cascades to table _iobeamdb_internal._hyper_1_1_0_partition
drop cascades to table _iobeamdb_internal._hyper_1_1_0_1_data
drop cascades to table _copy_temp_1
LOCATION:  reportDependentObjects, dependency.c:1011
drop tablespace tspace1;
